custom/icds_reports/tasks.py

```diff
@@ -1209,6 +1250,99 @@ def build_incentive_report(agg_date=None):
                 build_incentive_files.delay(location, agg_date, file_format, 3, location.parent.parent, location.parent)
 
 
+def is_launched_awc_incentive_report(performance_row, awcagg_row):
+    is_launched_from_awc = False
+    if awcagg_row[1].lower() == 'yes':
+        is_launched_from_awc = True
+    if performance_row[1] != is_launched_from_awc:
+        row_data = [performance_row[0], awcagg_row[0], performance_row[1], is_launched_from_awc]
+        return row_data
+    return None
+
+
+def eligibility_check_for_performance_report(performance_row):
+    if performance_row[2] in [0, None] or performance_row[3] in [0, None]:
+        is_eligible = True
+    else:
+        home_conduct_from_report = performance_row[2] / performance_row[3]
+        if home_conduct_from_report > 0.6:
+            is_eligible = True
+        else:
+            is_eligible = False
+    if is_eligible:
+        if performance_row[4] in [0, None] or performance_row[5] in [0, None]:
+            is_eligible = True
+        else:
+            weigh_eligibility = performance_row[4] / performance_row[5]
+            if weigh_eligibility > 0.6:
+                is_eligible = True
+            else:
+                is_eligible = False
+    if not performance_row[6]:
+        if is_eligible and performance_row[1]:
+            return [performance_row[0], is_eligible and performance_row[1], performance_row[6]]
+    else:
+        if not (is_eligible and performance_row[1]):
+            return [performance_row[0], is_eligible and performance_row[1], performance_row[6]]
+    return None
+
+
+def is_launched_incentive_check_report_helper(csv_data):
+    csv_columns = ['awc_id_from_performance', 'awc_id_from_aggregate', 'AwwIncentiveReport', 'AggAwc']
+    content = """
+    Please see attached file for is_launched mismatch in awc incentive report
+    """
+    _send_incentive_report_validation_email(content, csv_columns, csv_data)
+
+
+def home_conduct_awc_incentive_report(performance_row, awcagg_row):
+    if performance_row[2] in [0, None] or performance_row[3] in [0, None]:
```

Comment:
Code like this is very difficult to read. If there isn't a way to make it clearer, it would be helpful to have a comment or doc string describing what the expected input format is to help understand the intent of this. 

But probably it'd be better to extract it to a well-named method, e.g. `_performance_row_has_no_conduct_report` (again I am not sure about the names).

This comment applies many places in this PR

Reply:
okay

Reply:
changed